crates/extensions/tedge_flows/src/js_script.rs

```diff
@@ -116,24 +119,51 @@ impl JsScript {
         Ok(())
     }
 
+    /// Initialize the next execution time for this script's interval
+    /// Should be called after the script is loaded and interval is set
+    pub fn init_next_execution(&mut self) {
+        if !self.no_js_on_interval_fun && !self.interval.is_zero() {
+            self.next_execution = Some(Instant::now() + self.interval);
+        }
+    }
+
+    /// Check if this script should execute its interval function now
+    /// Returns true and updates next_execution if it's time to execute
+    pub fn should_execute_interval(&mut self) -> bool {
+        if self.no_js_on_interval_fun || self.interval.is_zero() {
+            return false;
+        }
+
+        let now = Instant::now();
+        match self.next_execution {
+            Some(deadline) if now >= deadline => {
+                // Time to execute - schedule next execution
+                self.next_execution = Some(now + self.interval);
+                true
+            }
+            None => {
+                // First execution - initialize and execute
+                self.next_execution = Some(now + self.interval);
+                true
+            }
+            _ => false,
+        }
+    }
+
     /// Trigger the onInterval function of the JS module
     ///
     /// The "onInterval" function is passed 2 arguments
     /// - the current timestamp
     /// - the current flow step config
     ///
     /// Return zero, one or more messages
+    ///
+    /// Note: Caller should check should_execute_interval() before calling this
     pub async fn on_interval(
```

Comment:
I see pros and cons to externalize or not the check if this call is timely. What are your motivation to move this check out?

Reply:
For the functions the javascript module exports for tedge-flows to call, my expectation is that the methods on `JsScript` serve as a lightweight wrapper, doing minimal work beyond calling the function with the provided input. It also feels like a bit of a hack to pretend the function produced no output when it wasn't called at all (or if it entirely doesn't exist).